mono/mini/local-propagation.c

```diff
@@ -368,12 +368,14 @@ mono_strength_reduction_ins (MonoCompile *cfg, MonoInst *ins, const char **spec)
 		}
 		break;
 	}
+#ifndef __s390__ // s390x microcode takes care of optimizing divide operations
 	case OP_IDIV_UN_IMM:
```

Comment:
We try to avoid platform specific ifdefs in this code. Why is this needed ?

Reply:
Would a runtime check be ok, like cfg->arch->optimize_divide?
i.e. is it preprocessor specifically, or any platform-conditionality, or unabstracted platform-conditionality?

Reply:
(personally I don't like preprocessor, because it breaks future configure/compile/link once to get target-aot-all)